src/main/java/com/appland/appmap/record/Recorder.java

```diff
@@ -154,6 +163,13 @@ public void add(Event event) {
     ts.isProcessing = true;
     try {
       if ( event.event.equals("call") ) {
+        if ( !ts.callStack.empty() && event.definedClass != null && AppMapConfig.get().isShallow(event.definedClass) ) {
+          Event parent = ts.callStack.peek();
+          if ( parent.definedClass != null && packageName(event.definedClass).equals(packageName(parent.definedClass)) ) {
```

Comment:
These calls to `packageName` are going to cause several allocations each time a method gets called. This will likely have a significant impact on performance. You should consider precomputing the package name in the constructor for `Event`.